assignments/lambda-classes.js

```diff
@@ -1 +1,134 @@
 // CODE here for your Lambda Classes
+
+class Person{
+    constructor(personAttributes){  
+    this.name = personAttributes.name;  
+    this.age = personAttributes.age;  
+    this.location = personAttributes.location  
+    }
+  
+    speak(){  
+     return `Hello, my name is ${this.name}, I am from ${this.location}.`;  
+     }  
+  }  
+     
+  class Instructor extends Person{  
+    constructor(instructorAttributes){  
+    super(instructorAttributes);  
+    this.specialty = instructorAttributes.specialty;  
+    this.favLanguage = instructorAttributes.favLanguage;  
+    this.catchPhrase = instructorAttributes.catchPhrase;  
+    
+      }
+     
+      demo(subject) {  
+   return `Today we are learning about ${subject}`;  
+    }
+     
+      grade(student,subject){  
+    return `${student.name} receives a perfect score on ${subject}`  
+     }
+    }
+  
+     
+  class Student extends Person{  
+    constructor(studentAttributes){  
+    super(studentAttributes);  
+    this.previousBackground = studentAttributes.previousBackground;  
+    this.className = studentAttributes.className;  
+    this.favSubjects = studentAttributes.favSubjects;  
+     }  
+    listsSubjects() {  
+    return `${this.favSubjects}`;  
+     }  
+    prAssignment(subject) {  
+    return `${this.name} has submitted a PR for ${subject}.`;  
+     }  
+    sprintChallenge() {  
+    return `${this.name} has begun a sprint challenge on ${this.className}.`;  
+   }  
+  }  
+     
+  class PM extends Instructor{  
+    constructor(pmAttributes){  
+    super(pmAttributes);  
+    this.gradClassName = pmAttributes.gradClassName;  
+    this.favInstructor = pmAttributes.favInstructor;  
+    }  
+    standUp(channel) {  
+      return `${this.name} announces to ${channel}, @channel standy times!`;  
+    }
+  
+    debugsCode(student, subject) {  
+      return `${this.name} debugs {student.name}'s code on ${subject}`;  
+    }  
```

Comment:
Good job passing the student object in here instead of simply passing the name parameter.